Fix part of #8472 . Migrate EditableTopicBackendApi service to Angular 8

adityasvat opened the pull request:
## Overview
<!--
READ ME FIRST: Please answer *both* questions below and check off every point from the essential checklist!
-->

1. This PR fixes or fixes part of #8472 .
2. This PR does the following: Migrate EditableTopicBackendApi service to Angular 8

## Essential Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The linter/Karma presubmit checks have passed locally on your machine.
- [x] "Allow edits from maintainers" is checked. (See [here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for instructions on how to enable it.)
  - This lets reviewers restart your CircleCI tests for you.
- [x] The PR is made from a branch that's **not** called "develop".

## PR Pointers

- Oppiabot will notify you when you don't add a PR_CHANGELOG label. If you are unable to do so, please @-mention a code owner (who will be in the Reviewers list), or ask on [Gitter](https://gitter.im/oppia/oppia-chat).
- For what code owners will expect, see the [Code Owner's wiki page](https://github.com/oppia/oppia/wiki/Oppia%27s-code-owners-and-checks-to-be-carried-out-by-developers).
- Make sure your PR follows conventions in the [style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide), otherwise this will lead to review delays
- Never force push. If you do, your PR will be closed.

adityasvat committed:
Added HTTP_INTERCEPTORS (topic-editor-page module)

adityasvat committed:
migrated EditableTopicBackendApiService

kevinlee12 committed:
Perform routine upgrade (#8943)

adityasvat committed:
fixed linting issues

adityasvat committed:
fixed linting issues

adityasvat commented:
@kevintab95 PTAL

jimbyo committed:
Fix part of #4057: Adds frontend test cases to QuestionObjectFactory (#8934)

* adding test coverage to QuestionObjectFactory

* Fixing linting errors

* fixing injector issue

* Adding More Test Coverage

* add QuestionObjectFactory to check_frontend_coverage.py

* add QuestionObjectFactory to check_frontend_coverage.py

* add QuestionObjectFactory to check_frontend_coverage.py

* fixing order

* fixing coding style issues

kevintab95 commented on core/templates/domain/topic/editable-topic-backend-api.service.ts:
```diff
@@ -16,180 +16,179 @@
-        });
-      }
-    };
+  updateTopic(topicId: string, topicVersion: string, commitMessage: string,
```
Looks like deleted a comment for this method. Can you please add that back?

kevintab95 commented on core/templates/domain/topic/editable-topic-backend-api.service.ts:
```diff
@@ -16,180 +16,179 @@
+  private _updateTopic(
+      topicId: string, topicVersion: string, commitMessage: string,
+      changeList: Array<Object>,
+      successCallback: any, errorCallback: any): any {
```
Ditto.

kevintab95 commented on core/templates/domain/topic/editable-topic-backend-api.service.ts:
```diff
@@ -16,180 +16,179 @@
+  }
+
+  private _deleteTopic(
+      topicId: string, successCallback: any, errorCallback: any): any {
```
ditto.

kevintab95 commented on core/templates/domain/topic/editable-topic-backend-api.service.ts:
```diff
@@ -16,180 +16,179 @@
-          errorCallback(errorResponse.data);
-        }
+  private _fetchSubtopicPage(
+      topicId: string, subtopicId: number, successCallback: any,
```
Avoid using the "any" type. Can you please take a look at other spec files to see what type can be used here?

kevintab95 commented on core/templates/domain/topic/editable-topic-backend-api.service.ts:
```diff
@@ -16,180 +16,179 @@
-          errorCallback(errorResponse.data);
-        }
+  private _fetchStories(
+      topicId: string, successCallback: any, errorCallback: any): any {
```
ditto.

kevintab95 commented on core/templates/domain/topic/editable-topic-backend-api.service.spec.ts:
```diff
@@ -224,86 +231,107 @@ describe('Editable topic backend API service', function() {
 
       expect(successHandler).not.toHaveBeenCalled();
-      expect(failHandler).toHaveBeenCalledWith('Error deleting topic 1.');
+      expect(failHandler).toHaveBeenCalled();
```
Use `toHaveBeenCalledWith`

kevintab95 commented on core/templates/domain/topic/editable-topic-backend-api.service.spec.ts:
```diff
@@ -224,86 +231,107 @@ describe('Editable topic backend API service', function() {
-    expect(failHandler).toHaveBeenCalledWith(
-      'Error loading story with id 0.');
-  });
+    expect(failHandler).toHaveBeenCalled();
```
Use `toHaveBeenCalledWith`.

kevintab95 commented on core/templates/domain/topic/editable-topic-backend-api.service.spec.ts:
```diff
@@ -224,86 +231,107 @@ describe('Editable topic backend API service', function() {
       var successHandler = jasmine.createSpy('success');
       var failHandler = jasmine.createSpy('fail');
 
       // Loading a topic the first time should fetch it from the backend.
```
Can you move this comment to the right place?

kevintab95 commented on core/templates/domain/topic/editable-topic-backend-api.service.spec.ts:
```diff
@@ -130,73 +121,87 @@ describe('Editable topic backend API service', function() {
       var failHandler = jasmine.createSpy('fail');
       var topic = null;
 
       // Loading a topic the first time should fetch it from the backend.
```
Can you move this comment to the right place?

kevintab95 commented on core/templates/domain/topic/editable-topic-backend-api.service.spec.ts:
```diff
@@ -16,110 +16,101 @@
+import { HttpClientTestingModule, HttpTestingController } from
+  '@angular/common/http/testing';
+import { TestBed, fakeAsync, flushMicrotasks } from '@angular/core/testing';
+import {EditableTopicBackendApiService} from
```
```suggestion
import { EditableTopicBackendApiService } from
```

kevintab95 commented on core/templates/domain/topic/editable-topic-backend-api.service.spec.ts:
```diff
@@ -16,110 +16,101 @@
+import { TestBed, fakeAsync, flushMicrotasks } from '@angular/core/testing';
+import {EditableTopicBackendApiService} from
+  'domain/topic/editable-topic-backend-api.service.ts';
+import {CsrfTokenService} from 'services/csrf-token.service.ts';
```
```suggestion
import { CsrfTokenService } from 'services/csrf-token.service.ts';
```

kevintab95 commented on core/templates/domain/topic/editable-topic-backend-api.service.spec.ts:
```diff
@@ -16,110 +16,101 @@
   it('should successfully fetch an existing topic from the backend',
-    function() {
+    fakeAsync(() => {
       var successHandler = jasmine.createSpy('success');
```
Use `let` here.

Ditto below in this file.

kevintab95 commented on core/templates/domain/topic/editable-topic-backend-api.service.spec.ts:
```diff
@@ -130,73 +121,87 @@ describe('Editable topic backend API service', function() {
 
       expect(successHandler).not.toHaveBeenCalled();
-      expect(failHandler).toHaveBeenCalledWith('Error loading topic 1.');
+      expect(failHandler).toHaveBeenCalled();
```
Use `toHaveBeenCalledWith` like before.

kevintab95 commented on core/templates/domain/topic/editable-topic-backend-api.service.spec.ts:
```diff
@@ -130,73 +121,87 @@ describe('Editable topic backend API service', function() {
     expect(successHandler).not.toHaveBeenCalled();
-    expect(failHandler).toHaveBeenCalledWith('Error loading subtopic 1.');
-  });
+    expect(failHandler).toHaveBeenCalled();
```
Use `toHaveBeenCalledWith`.

kevintab95 reviewed (changes requested):
Thanks @adityalpha! I've taken a pass and left some comments. PTAL.

Also, can you please merge from develop when you address the comments? This is just to ensure that there aren't any issues after your PR goes in. Thanks!

kevintab95 commented:
Please make sure to assign the reviewer once your PR is ready for review. Thanks!

bansalnitish commented:
Hey @kevintab95, please assign me once you're fine with the PR.

nishantwrp committed:
Remove any from url-interpolation.service.ts, url.service.ts and some other files (#8823)

* Remove any from some other files

* Remove compilation errors

* Remove any from some more files

* Review

* Add Comments

* Add Comments

* Add proper comments

* make -> makes

srijanreddy98 committed:
Fixes not loading opportunities in the community dashboard. (#8948)

* Migrated contribution-opportunities-backend-api.service.ts from AngularJS to Angular8. Migrate AngularJS files to Angular 8 #8472

* Migrated contribution-opportunities-backend-api.service.ts from AngularJS to Angular8. Migrate AngularJS files to Angular 8 #8472

* Migrated contribution-opportunities-backend-api.service.ts from AngularJS to Angular8. Migrate AngularJS files to Angular 8 #8472

* Migrated contribution-opportunities-backend-api.service.ts from AngularJS to Angular8. Migrate AngularJS files to Angular 8 #8472

* Refactor some functions

* Update contribution-opportunities-backend-api.service.ts

* Resolve Merge Conflicts

* Revert yarn.lock

* Revert "Resolve Merge Conflicts"

This reverts commit b974e7831bf6eb56c2b7c26fa26da39031d41bc8.

* Update contribution-opportunities-backend-api.service.ts

* Added spec file for ContributionOpportunitiesBackendApi service

* Added tests for fetch tanslation and fetch voiceover opportunitites

* Update contribution-opportunities-backend-api.service.ts

Remove extra space after the last function argument.

* Added type for cursor

* Change fdescribe and fit back to describe and it

* Code review changes

* Code review changes

* Code review changes

* Fixing typos

* Remove Skill Domain Types from this pr

* Code review

* Code review

* Code review

* Fix not loading contribution opportunities

marianadasilvadev committed:
Fix #8920 adding check frontend coverage feature on pre push hook (#8923)

* Add --check_coverage flag in run_frontend_tests scripts on pre push hook

* Refactor function from pre_push_hook

* Fix review comments

* Fix pre_push_hook

* Add typeinfo

vojtechjelinek committed:
Upgrade major versions of libraries (#8953)

* Perform routine upgrade for yarn packages

* upgrade webpack packages to newer major versions

* Upgrade karma packages to newer major versions

* Move Angular 8 dependencies from devDependencies

* Update yarn lock

* Fix frontend tests

* Fix frontend tests

* Fix e2e tests

ankita240796 committed:
Fix issues with release wrap up (#8952)

DubeySandeep committed:
Fixes reported issues in community dashboard page (#8939)

* Fixes reported issues.

* Resolved minor issues.

* Added correct success message.

* Adds rootScope.apply.

* rootscope apply for voicoever opportunities.

* Adds margin in progress percentage.

brianrodri committed:
Fix #8955: Only fetch the summaries of threads referring to explorations (#8956)

* Only fetch the summaries of threads referring to an exploration.

* fix line-too-long

* fix typo in docstring

bansalnitish committed:
Adds http interceptor in all the modules (#8965)

felicityzhao99 committed:
Fix issue #4057: Add front-end tests for UnitsObjectFactory.ts (#8915)

* Add front-end tests for UnitsObjectFactory.ts

* Add front-end tests for UnitsObjectFactory.ts

* Add front-end tests for UnitsObjectFactory.ts

* Add front-end tests for UnitsObjectFactory.ts

* Add front-end tests for UnitsObjectFactory.ts

* fix the Module not found bug

* Change the format of codes

* change the format

* Rerun to check Travis-ci test

* comma problem

kevinlee12 committed:
Routine upgrade yarn packages (#8969)

DubeySandeep committed:
Fix #7402 & #8166: Enables community dashboard page (#8957)

* Enables community dashboard.

* Serve community dashboard page statically.

* Fixes community dashboard page prod issue.

* Resolves review comments.

brianrodri committed:
Fix InteractionAnswerSummariesMRJobManager handling of utf-8 strings (#8963)

* decode utf-8 before splitting

* remove unused import

* add problematic state names from test server

* fix test name

* only keep test for valid unicode

* remove unreachable except handler

bansalnitish commented:
@adityalpha any updates on this PR? Let me know if you need any help.

iamprayush committed:
Fix #8926: Drag and drop response order  (#8960)

* Added selectedRank in the choices object.

* Added default initialization of selectedRanks

* Fixed default initialization of selectedRanks

* Fixed default initialization of selectedRanks.

* Moving initializations outside the inner loop.

DubeySandeep committed:
Fix UI issues in community dashboard page (#8967)

h-ARTS committed:
Migration of ExpressionSyntaxTreeService to Angular 8 (#8886)

* ExpressionSyntaxTreeService added to UpgradedServices

* Migrate ExpressionSyntaxTreeService to Angular 8

* Minor changes:

- privatised method verifyNumArgs
- TestBed used for Service
- Lines coerced for verifyNumArgs

* Adjusted arguments in one line after autoformat issue

* Placed exact types for evaluation and getter method.

- short-circuit comment removed as it's been already done using logical/ternary operators

* Math.pow.eval number args fix by using the numericArgs variable and type corrections

* underscores removed from all methods

* Expression error types converted to classes

* Extracted recursive findParams private method

* Inverted condition from logical AND to OR

* a >= b && c <= d    to     a < b || c > d

* removed name property and changes to constructor

- constructor left blank
- workaround for restoring prototype chain

* Some minor changes to the types and return statement of getParamsUsedInExpression

* quotation marks added to if, floor, pow, log and abs

* findParams return changed to Set data structure

* Revert "findParams return changed to Set data structure"

This reverts commit 7cef239102bbfa800aae35372a3c7f100f97d9b7.

* TODO Issue added

* Tests for expression errors

* typo on exprUndefinedVarError

* ExpressionSyntaxTreeService placed in upgraded service in topological level 1

ankita240796 committed:
Move all wrap up steps to script (#8974)

* Move all wrap up steps to script

* Update function names

adityasvat committed:
resolved merge remaining conflicts
